scala-package/core/src/main/scala/ml/dmlc/mxnet/Visualization.scala

```diff
@@ -0,0 +1,277 @@
+      */
+    def edge(tailName: String, headName: String,
+        label: String = null, attrs: Map[String, String]): Unit = {
+      _body :+= _edge.format(tailName, headName, attribute(label, attrs))
+    }
+
+    private def save(filename: String, directory: String): String = {
+      val path = s"$directory${File.separator}$filename"
+      val writer = new PrintWriter(path)
+      writer.write(s"${this._head}\n")
+      this._body.foreach { line => writer.write(s"$line\n") }
+      writer.write(s"${this._tail}\n")
+      writer.flush()
+      writer.close()
+      path
+    }
+
+    private def command(engine: String, format: String, filepath: String): String = {
+      assert(ENGINES.contains(engine) == true, s"unknown engine: $engine")
+      assert(FORMATS.contains(format) == true, s"unknown format: $format")
+      s"$engine -T${format} -O $filepath"
+    }
+
+    /**
+     * Render file with Graphviz engine into format.
+     *  @param engine The layout commmand used for rendering ('dot', 'neato', ...).
+     *  @param format The output format used for rendering ('pdf', 'png', ...).
+     * @param fileName Name of the DOT source file to render.
+     * @param path Path to save the Dot source file.
+     */
+    def render(engine: String = "dot", format: String = "pdf",
+        fileName: String, path: String): Unit = {
+      val filePath = this.save(fileName, path)
+      val args = command(engine, format, filePath)
+      import sys.process._
+      try {
+        args !
+      } catch { case _ : Throwable =>
+        val errorMsg = s"""failed to execute "$args", """ +
+        """"make sure the Graphviz executables are on your systems' path"""
+        throw new RuntimeException(errorMsg)
+      }
+    }
+  }
+
+  /**
+   * convert shape string to list, internal use only
+   * @param str shape string
+   * @return list of string to represent shape
+   */
+  def str2Tuple(str: String): List[String] = {
+    val re = """\d+""".r
+    re.findAllIn(str).toList
+  }
+
+  /**
+   * convert symbol to Dot object for visualization
+   * @param symbol symbol to be visualized
+   * @param title title of the dot graph
+   * @param shape Map of shapes, str -> shape, given input shapes
+   * @param nodeAttrs Map of node's attributes
+   *               for example:
+   *                      nodeAttrs = Map("shape" -> "oval", "fixedsize" -> "fasle")
+   *                      means to plot the network in "oval"
+   * @return Dot object of symbol
+   */
+  def plotNetwork(symbol: Symbol, title: String = "plot",
+      saveFormat: String = "pdf", shape: Map[String, Shape] = null,
+      nodeAttrs: Map[String, String] = Map[String, String]()): Dot = {
+
+    val (drawShape, shapeDict) = {
+      if (shape == null) (false, null)
+      else {
+        val internals = symbol.getInternals()
+        val (_, outShapes, _) = internals.inferShape(shape)
+        assert(outShapes != null, "Input shape is incompete")
```

Comment:
assert -> require, so as the others
assert can be disabled
http://stackoverflow.com/questions/26140757/what-to-choose-between-require-and-assert-in-scala